### ISSUE TITLE
Replace `jsonrpc` in `zebra-rpc`

### ISSUE THREAD

**Issue by upbqdn** [ZcashFoundation/zebra#8682]
`zebra-rpc` uses https://github.com/paritytech/jsonrpc as its JSON RPC server. However, the crate is no longer actively developed and has been superseded by https://github.com/paritytech/jsonrpsee. Another option would be https://github.com/google/tarpc.

**Comment by conradoplg** [comment]
I seem to recall that zcashd uses a mix of JSON-RPC 1.0 and 2.0 and that we had to add some workarounds, so we'll need to check if one of those crates are compatible

**Comment by arya2** [comment]
It looks like `jsonrpsee` supports middleware via [set_http_middleware](https://docs.rs/jsonrpsee-server/0.24.0/jsonrpsee_server/struct.ServerBuilder.html#method.set_http_middleware)/[set_rpc_middleware](https://docs.rs/jsonrpsee-server/0.24.0/jsonrpsee_server/struct.ServerBuilder.html#method.set_rpc_middleware), though they require different traits.

We should try using tonic too, it seems more extensible if we get it working with a [JSON codec](https://github.com/ZcashFoundation/zebra/issues/8245) .

**Comment by upbqdn** [comment]
I see having a single RPC framework as beneficial, so I'd go with tonic as well.

**Comment by mpguerra** [comment]
blocked on #9006 and #9049
